### DISCUSSION TITLE
Should I call check_active_accounts for each configuration

### DISCUSSION THREAD

**Discussion by jesseduffield** [janko/rodauth-rails#298] [Q&A] [answered]
I've got two configurations here, main and admin. I'm checking active sessions in both but I worry that's inefficient or unnecessary: Is this what I'm supposed to be doing? And if so, does it require two DB queries per request, or will it only be one? Thanks

```ruby
class RodauthApp < Rodauth::Rails::App
  configure RodauthMain
  configure RodauthAdmin, :admin

  route do |r|
    rodauth.check_active_session
    rodauth(:admin).check_active_session

    r.rodauth
    r.rodauth(:admin)
  end
end
```

**Comment by janko** [comment, marked as answer]
You need both calls if the main and admin configurations differ in `session_key_prefix` or account record retrieval (different tables or distinct table subsets). If your main and admin configurations share the session and table dataset, then it might not be technically necessary, but IMO it's still more correct to have both.

**Reply by jesseduffield** [reply]
They do differ in that I'm doing:
```ruby
  def account_table_ds
    super.where(type: account_type)
  end
```
where account_type is different in both configurations. At any rate, I'm happy to just have both, thanks!
